### PR TITLE
fix(@schematics/angular): handle aliased or existing environment import

### PR DESCRIPTION
**Pull request by arturovt**
Closes: #16226

The exception `Cannot read property 'length' of undefined` wasn't really informative thus I had to find out where this exception was actually thrown. All ways led me to the `Chunk` class and the `insertLeft` method that used `buffer`'s variable `length` property. The `buffer` was `undefined`, that's why that exception was thrown. It was `undefined` because there is no way to check if `environment` is already imported.

This PR adds the ability to handle the already existing "environment" import or the aliased one.

### CODE REVIEW

File: packages/schematics/angular/utility/ast-utils.ts [hunk 1]
```diff
@@ -574,6 +574,59 @@ export function isImported(source: ts.SourceFile,
...
+    )
+    .find((declaration: ts.ImportDeclaration) => {
+      if (!declaration.importClause) {
+        return false;
+      }
+
+      // Find specifier that contains `environment` keyword in its text.
+      // Whether it's `environment` or `environment as env`.
+      const nodes = findNodes(declaration, ts.SyntaxKind.ImportSpecifier).filter(node =>
+        node.getText().includes('environment'),
```
Comment (clydin) [inline packages/schematics/angular/utility/ast-utils.ts]: `getText` can be an expensive operation.
Can you move the named bindings analysis up to this point and use it to check the values?

Comment (arturovt) [reply packages/schematics/angular/utility/ast-utils.ts]: I hope I understand you correctly. :slightly_smiling_face:

**Comment by clydin** [comment]
Can you rebase on master?  This should fix the failing tests.

**Comment by arturovt** [comment]
> Can you rebase on master? This should fix the failing tests.

Done. `setup-and-build-win` job seems to be happy:)

**Pull request merged**